src/main/java/com/limechain/storage/block/BlockState.java

```diff
@@ -0,0 +1,796 @@
+        BlockHeader header = getHeaderByNumber(blockNum);
+
+        if (header == null) {
+            throw new RuntimeException("Header not found for block number: " + blockNum);
+        }
+
+        return getAllBlocksAtDepth(header.getParentHash().getBytes());
+    }
+
+    /**
+     * Gets all the hashes of unfinalized blocks with the depth of the given hash plus one
+     *
+     * @param hash the parent hash
+     * @return List of hashes of blocks
+     */
+    public List<byte[]> getAllBlocksAtDepth(byte[] hash) {
+        return blockTree.getAllBlocksAtNumber(hash);
+    }
+
+    /**
+     * Verifies if a block is on our current chain by checking if it is a descendant of our best block
+     *
+     * @param header the block header
+     * @return true if the block is on our current chain, false otherwise
+     */
+    public boolean isBlockOnCurrentChain(BlockHeader header) {
+        BlockHeader bestBlock = bestBlockHeader();
+        if (bestBlock == null) {
+            throw new RuntimeException("Best block header cannot be retrieved");
+        }
+
+        // If the new block's number is greater than our best block's number, then it is on our current chain.
+        if (header.getBlockNumber().compareTo(bestBlock.getBlockNumber()) > 0) {
+            return true;
+        }
+
+        return isDescendantOf(header.getHash(), bestBlock.getHash());
+    }
+
+    /**
+     * Gets the current beest block's hash
+     *
+     * @return the best block's hash
+     */
+    public byte[] bestBlockHash() {
+        return blockTree.bestBlockHash();
+    }
+
+    /**
+     * Gets the current best block's header
+     *
+     * @return the best block's header
+     */
+    public BlockHeader bestBlockHeader() {
+        byte[] bestHash = bestBlockHash();
+        return getHeader(bestHash);
+    }
+
+    /**
+     * Gets the current best block's state root's hash
+     *
+     * @return the best block's state root's hash in byte array representation
+     */
+    public byte[] bestBlockStateRoot() {
+        return bestBlockHeader().getStateRoot().getBytes();
+    }
+
+    /**
+     * Gets the given block's state root's hash
+     *
+     * @param blockHash the block hash
+     * @return the given block's state root's hash in byte array representation
+     */
+    public byte[] getBlockStateRoot(byte[] blockHash) {
+        return getHeader(blockHash).getStateRoot().getBytes();
+    }
+
+    /**
+     * Gets the current best block's number
+     *
+     * @return the best block's number
+     */
+    public BigInteger bestBlockNumber() {
+        BlockHeader header = bestBlockHeader();
+        if (header == null) {
+            throw new RuntimeException("Failed to get best block header");
+        }
+        return header.getBlockNumber();
+    }
+
+    /**
+     * Gets the current best block
+     *
+     * @return the best block
+     */
+    public Block bestBlock() {
+        return getBlockByHash(bestBlockHash());
+    }
```

Comment:
Is it the best one finalized or non-finalized one? I think it's worth mentioning in the javadoc